conans/test/functional/tools/system/package_manager_test.py

```diff
@@ -30,11 +29,51 @@ def system_requirements(self):
 
 @pytest.mark.tool_apt_get
 @pytest.mark.skipif(platform.system() != "Linux", reason="Requires apt")
-@pytest.mark.skipif(six.PY2, reason="Does not pass on Py2 with Pytest")
+def test_apt_install_substitutes():
+    client = TestClient()
+    conanfile_py = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.system.package_manager import Apt
+        class MyPkg(ConanFile):
+            settings = "arch", "os"
+            def system_requirements(self):
+                # FIXME this is needed because the ci-functional apt-get update fails
+                try:
+                    self.run("sudo apt-get update")
+                except Exception:
+                    pass
+                apt = Apt(self)
+                {}
+        """)
+
+    client.run("remove test/1.0 -f")
+
+    installs = 'apt.install_substitutes(["non-existing1", "non-existing2"], ["non-existing3", "non-existing4"])'
+    client.save({"conanfile.py": conanfile_py.format(installs)})
+    client.run("create . test/1.0@ -c tools.system.package_manager:mode=install "
+               "-c tools.system.package_manager:sudo=True", assert_error=True)
+    assert "dpkg-query: no packages found matching non-existing1:amd64" in client.out
+    assert "dpkg-query: no packages found matching non-existing2:amd64" in client.out
+    assert "dpkg-query: no packages found matching non-existing3:amd64" in client.out
+    assert "dpkg-query: no packages found matching non-existing4:amd64" in client.out
+    assert "ERROR: while executing system_requirements(): " \
+           "None of the installs for the package substitutes succeeded." in client.out
+
+    client.run_command("sudo apt remove nano -yy")
+    installs = 'apt.install_substitutes(["non-existing1", "non-existing2"], ["nano"], ["non-existing3"])'
```

Comment:
I think it could be better to have one-failing and one-succeeding in the same list:

```suggestion
    installs = 'apt.install_substitutes(["non-existing1", "non-existing2"], ["non-existing3", "nano"])'
```

Reply:
That would fail, you need at least one list of all valid packages